pandas/tests/io/pytables/test_store.py

```diff
@@ -84,28 +82,26 @@ def test_format_kwarg_in_constructor(self, setup_path):
                 HDFStore(path, format="table")
 
     def test_context(self, setup_path):
-        path = create_tempfile(setup_path)
-        try:
-            with HDFStore(path) as tbl:
-                raise ValueError("blah")
-        except ValueError:
-            pass
-        finally:
-            safe_remove(path)
-
-        try:
-            with HDFStore(path) as tbl:
-                tbl["a"] = tm.makeDataFrame()
-
-            with HDFStore(path) as tbl:
-                assert len(tbl) == 1
-                assert type(tbl["a"]) == DataFrame
-        finally:
-            safe_remove(path)
+        with tm.ensure_clean() as path:
+            try:
+                with HDFStore(path) as tbl:
+                    raise ValueError("blah")
+            except ValueError:
+                pass
+            try:
+                with HDFStore(path) as tbl:
+                    tbl["a"] = tm.makeDataFrame()
+            except ValueError:
+                pass
+            try:
+                with HDFStore(path) as tbl:
+                    assert len(tbl) == 1
+                    assert type(tbl["a"]) == DataFrame
+            except ValueError:
+                pass
```

Comment:
sure we want to pass here and on L95?

Reply:
Haven't modified the existing structure here. What would you suggest to change it to?

Reply:
the `except ValueError: pass` on L100-101 and L94-95 are new